platform/core.multitabs/src/org/netbeans/core/multitabs/impl/ProjectColorTabDecorator.java

```diff
@@ -63,15 +62,27 @@ public void stateChanged(ChangeEvent e) {
 
     static {
         backGroundColors = new ArrayList<Color>( 10 );
-        backGroundColors.add( new Color( 216, 255, 237 ) );
-        backGroundColors.add( new Color( 255, 221, 221 ) );
-        backGroundColors.add( new Color( 255, 247, 214 ) );
-        backGroundColors.add( new Color( 216, 239, 255 ) );
-        backGroundColors.add( new Color( 241, 255, 209 ) );
-        backGroundColors.add( new Color( 255, 225, 209 ) );
-        backGroundColors.add( new Color( 228, 255, 216 ) );
-        backGroundColors.add( new Color( 227, 255, 158 ) );
-        backGroundColors.add( new Color( 238, 209, 255 ) );
+
+        // load background colors from UI defaults if available
+        if (UIManager.getColor("nb.multitabs.project.1.background") != null) {
+            for (int i = 1; i <= 100; i++) {
+                Color color = UIManager.getColor("nb.multitabs.project." + i + ".background");
```

Comment:
Is this a gradient of some sort? Better to specify start and end colors, and interpolate, than have up to 100 properties. That will look better on HiDPI displays too (since you can actually get a smooth gradient rather than N hard steps).

Reply:
And you probably don't need to do the interpolation yourself... there's Graphics2D.setPaint(new GradientPaint(...))

Reply:
> Is this a gradient of some sort?

No, NetBeans has an option "same background color for files from same project" that paints tabs in different colors. Each open project gets its own color. The used colors are hardcoded into `ProjectColorTabDecorator` and work good with light LAFs, but not with dark LAFs. So I've added a way to get those colors from [UI defaults](https://github.com/DevCharly/netbeans/blob/d3c9d24efe5b8e48b19c91795c19e99354d8c9bd/platform/o.n.swing.laf.flatlaf/src/org/netbeans/swing/laf/flatlaf/FlatDarkLaf.properties#L68-L76).

Reply:
Ah, makes sense.